src/deepsparse/transformers/pipelines/text_generation.py

```diff
@@ -448,14 +465,40 @@ def engine_forward(
         with self.timer_manager.new_timer_context(total_inference=False) as timer:
             streamer = context.get("streamer")
 
+            # engine_inputs is a list of numpy arrays plus additional
+            # session_id string. We need to pop the session_id string
+            # and from the engine_inputs. The session_id will be used
+            # seperately to keep track of the appropriate kv cache session
+            # (if kv cache is enabled)
+            session_id = engine_inputs.pop(
```

Comment:
Sara: what if there is another string that we are adding. 
Better: engine_inputs.pop(-1)